DB/src/test/java/io/deephaven/db/v2/utils/TestKeyedArrayBackedMutableTable.java

```diff
@@ -278,10 +267,24 @@ private void waitForCompletion() throws InterruptedException, ExecutionException
         }
     }
 
-    private void handleDelayedRefresh(BaseArrayBackedMutableTable kabut) {
-        new Thread(() -> {
-            SleepUtil.sleep(200);
-            LiveTableMonitor.DEFAULT.runWithinUnitTestCycle(kabut::refresh);
-        }).start();
+    private void handleDelayedRefresh(final BaseArrayBackedMutableTable table, final FunctionalInterfaces.ThrowingRunnable<IOException> action) throws Exception {
+        final Thread refreshThread;
+        final Semaphore gate = new Semaphore(0);
```

Comment:
I think CountDownLatch is more appropriate construct to user here.

Reply:
It's functionally equivalent, but sure, it's a better name. I had considered it, but didn't use it just because it's slightly more obscure. I'll change it.

Reply:
This is done. While doing it, I noticed a way to improve the concurrency of the test, so that's a nice side effect.